File: src/main/java/practice/user/application/in/UserService.java
package practice.user.application.in;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import practice.adaptor.out.entity.UserEntity;
import practice.user.application.out.LoadUserPort;
import practice.user.application.out.RegisterUserPort;
import practice.user.domain.User;

@Service
@RequiredArgsConstructor
public class UserService implements RegisterUserUseCase {

  private final LoadUserPort loadUserPort;
  private final RegisterUserPort registerUserPort;


  @Override
  public boolean alreadyExists(String name, String email) {
    return loadUserPort.hasUser(name, email);
  }

  @Override
  public User register(String name, String email) {
    return registerUserPort.register(name, email);
  }
}

File: src/main/kotlin/practice/user/application/in/UserService.kt
package practice.user.application.`in`

import org.springframework.stereotype.Service
import practice.user.application.out.LoadUserPort
import practice.user.application.out.RegisterUserPort
import practice.user.domain.User

@Service
class UserService(
    private val loadUserPort: LoadUserPort,
    private val registerUserPort: RegisterUserPort
) : RegisterUserUseCase {


    override fun alreadyExists(name: String, email: String): Boolean {
        return loadUserPort.hasUser(name, email)
    }

    override fun register(name: String, email: String): User {
        return registerUserPort.register(name, email)
    }
}
